[Chong Yao Wei] iP

yaowei-soc opened the pull request.

yaowei-soc committed:
Level 1 complete

yaowei-soc committed:
Level 2 complete

yaowei-soc committed:
Level 3 complete

yaowei-soc committed:
Level 4 complete

yaowei-soc committed:
Fix array index

yaowei-soc committed:
TextUiTesting complete

yaowei-soc committed:
Level 5 complete

yaowei-soc committed:
Level 6 complete

yaowei-soc committed:
Added basic Javadocs for classes

yaowei-soc committed:
Added date parsing method

Parses date for deadline and event commands

yaowei-soc committed:
refactor: move code into packages and classes

- Made use of Command pattern to process the user inputs
- Design is heavily inspired from
https://github.com/se-edu/addressbook-level2
- From this commit onwards, I will be attempting to follow
https://www.conventionalcommits.org/en/v1.0.0/

yaowei-soc committed:
feat: save tasks to disk

Tasks are saved to disk on every command and loaded when application is
launched.

yaowei-soc committed:
Merge branch 'level-7'

yaowei-soc committed:
Merge branch 'level-8'

yaowei-soc committed:
style: added Intellij editor code style scheme

The base config is from
https://github.com/google/styleguide/blob/gh-pages/intellij-java-google-style.xml
with additional changes from
https://se-education.org/guides/conventions/java/basic.htmlhttps://se-education.org/guides/conventions/java/basic.html

yaowei-soc committed:
feat: added basic junit tests

Run tests using Gradle

yaowei-soc committed:
build: configure gradle to produce jar file

yaowei-soc committed:
feat: implemented find command

find is implemented using String.contains

yaowei-soc committed:
style(checktyle): add checkstyle config

checkstyle config file is extracted from  se-edu /
addressbook-level4 with edits to ensure that it works with the current
version of checkstyle

The modules changed are
- LineLength
- allowedAnnotations
- allowUndeclaredRTE
- allowMissingPropertyJavadoc
- allowMissingThrowsTags
- minLineCount

- Add google checkstyle and suppression config
- Formatted code

Additionally, source files are now under src/main/java for gradle to
detect.

yaowei-soc committed:
feat: add javafx in gradle dependencies

Add basic class to display a GUI popup to test if JavaFX works

QY-H00 commented on src/main/com/nus/duke/data/Searchable.java:
```diff
@@ -0,0 +1,6 @@
+package com.nus.duke.data;
+
+public interface Searchable {
```
Will it be better to add a JavaDoc here to desribe Searchable? I'm a bit confused about this interface

QY-H00 replied:
Maybe is it better to add JavaDoc here to help decribe Searchable?

QY-H00 reviewed (commented):
I have studied from your code and it looks pretty nice overall : )

QY-H00 commented on src/main/com/nus/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,51 @@
+import com.nus.duke.data.Deadline;
+import com.nus.duke.parser.DateParser;
+
+import java.util.regex.Matcher;
```
Should this import sentence be put above importing our own class？

QY-H00 commented on src/main/com/nus/duke/command/EventCommand.java:
```diff
@@ -0,0 +1,52 @@
+import com.nus.duke.data.Event;
+import com.nus.duke.parser.DateParser;
+
+import java.util.regex.Matcher;
```
Same as above.

QY-H00 commented on src/main/com/nus/duke/parser/CommandParser.java:
```diff
@@ -0,0 +1,58 @@
+import com.nus.duke.command.IncorrectCommand;
+import com.nus.duke.command.ListCommand;
+import com.nus.duke.command.TodoCommand;
+import java.util.regex.Matcher;
```
Same advice as above for coding standard regarding to import.

QY-H00 commented on src/main/com/nus/duke/storage/TaskListStorage.java:
```diff
@@ -0,0 +1,41 @@
+import com.nus.duke.data.Task;
+import com.nus.duke.data.TaskList;
+
+import java.io.*;
```
Should we use explicit import instead of wild card import?

QY-H00 commented on src/test/java/com/nus/duke/parser/CommandParserTest.java:
```diff
@@ -0,0 +1,121 @@
+import com.nus.duke.command.IncorrectCommand;
+import com.nus.duke.command.ListCommand;
+import com.nus.duke.command.TodoCommand;
+import org.junit.jupiter.api.BeforeEach;
```
Is it better if we leave a line between different importing group?

QY-H00 reviewed (commented):
For coding standard, LGTM. Maybe it's better to follow the standard of importing.

QY-H00 reviewed (commented):
For coding quality, I appreciate your use of OOP feature such as polymorphism and inheritance! Maybe it's better to add some JavaDoc to defined class to help other developer understand.

yaowei-soc committed:
feat: add graphical ui using javafx

yaowei-soc committed:
Merge branch 'level-10'

Add basic GUI support for displaying and entering commands

yaowei-soc committed:
style(checkstyle): fix various style violations

yaowei-soc committed:
feat: better interface

Followed SE EDU guides on JavaFX ui

yaowei-soc committed:
test: add assert statements

yaowei-soc committed:
refactor: code quality changes

yaowei-soc committed:
Merge pull request #2 from yaowei-soc/better-gui

Better GUI

yaowei-soc committed:
ci: add github actions file from se-edu

yaowei-soc committed:
ci: remove text-ui-test

Removed text-ui-text for CI since it is not being used

yaowei-soc committed:
build(gradle): update jar file name

shadowJar should now create a jar file with the name 'duke'

yaowei-soc committed:
docs: add basic user guide

User guide template is based on address book level 4 format

yaowei-soc committed:
docs: update README.md

update readme to link to user guide

yaowei-soc committed:
docs: add date time formatting section

Add date/time formatting documentation

yaowei-soc committed:
feat: set default path for storage data

Set the default storage path to the home directory for cross platform compatibility

yaowei-soc committed:
Set theme jekyll-theme-cayman

yaowei-soc committed:
feat: better gui

Added dark theme and better display for message dialog

yaowei-soc committed:
docs: update javadocs

Added javadocs for existing classes

yaowei-soc committed:
docs: update screenshot to reflect latest ui update

yaowei-soc committed:
test: add missing deadline and event tests

yaowei-soc committed:
build: update build.gradle

yaowei-soc committed:
fix: vbox alignment for dialog boxes

New dialog boxes are now pinned to the top instead of the center of the main window

yaowei-soc committed:
build: fix no toolkit found error in ubuntu

yaowei-soc committed:
fix: updated dateparser

- Updated date parser to handle more date and time combinations
- Updated DateParserTest

yaowei-soc committed:
fix: javafx version warning

yaowei-soc committed:
refactor: renamed Main.java to Launcher.java

yaowei-soc committed:
docs: update README.md

yaowei-soc committed:
docs: update javadoc for TaskListStorage.java

yaowei-soc committed:
refactor: coding standard formatting

yaowei-soc committed:
Merge pull request #3 from yaowei-soc/branch-A-JavaDoc

Update javadoc for TaskListStorage.java

yaowei-soc committed:
Merge pull request #4 from yaowei-soc/branch-A-CodingStandard

Coding standard formatting

yaowei-soc committed:
dummy commit for branch-A-CodeQuality

yaowei-soc committed:
dummy commit for branch-A-Assertions

yaowei-soc committed:
Merge pull request #5 from yaowei-soc/branch-A-Assertions

dummy commit for branch-A-Assertions

yaowei-soc committed:
Merge pull request #6 from yaowei-soc/branch-A-CodeQuality

dummy commit for branch-A-CodeQuality
